### PR TITLE
feat: move collapsible partition forms to side panel

### PR DESCRIPTION
**Pull request by Jay-Topher**
## Done
Moved partition forms to side panel:
- Edit partition
- Create cache set
- Remove partition
- Create Bcache
- Create Logical Volume
- Delete Volume Group

## QA steps

- [ ] Visit the storage tab of a ready state machine
- [ ] Create a partition from any of the available disks
- [ ] Ensure that the forms listed above display in a side panel
- [ ] Ensure that the actions work as expected

## Fixes

Fixes: [Move partition forms to side panel](https://warthogs.atlassian.net/browse/MAASENG-2873)

### Before
image
image
image

### After
image
image
image

### CODE REVIEW

File: src/app/base/components/node/StorageTables/AvailableStorageTable/DeletePartition/DeletePartition.test.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    />,
+    { state }
+  );
+
+  expect(
+    screen.getByRole("form", { name: "Delete partition" })
+  ).toBeInTheDocument();
+});
+
+it("should fire an action to delete a disk", async () => {
```
Comment (petermakowski) [inline src/app/base/components/node/StorageTables/AvailableStorageTable/DeletePartition/DeletePartition.test.tsx:49]: The action that you check within this test is for deleting a partition, not a disk.

File: src/app/base/components/node/StorageTables/AvailableStorageTable/AvailableStorageTable.test.tsx [hunk 2]
```diff
@@ -258,12 +265,7 @@ describe("performing machine actions", () => {
   });
 
   it("can open the add partition form if disk can be partitioned", async () => {
-    vi.spyOn(sidePanelHooks, "useSidePanel").mockReturnValue({
-      setSidePanelContent,
-      sidePanelContent: null,
-      setSidePanelSize: vi.fn(),
-      sidePanelSize: "regular",
-    });
+    mockSidePanelContent();
```
Comment (petermakowski) [inline src/app/base/components/node/StorageTables/AvailableStorageTable/AvailableStorageTable.test.tsx:268]: I'd suggest to move this spy to `beforeEach` instead.

**Review by petermakowski — COMMENTED** [review]
Code looks solid, haven't been able to QA just yet.

**Comment by petermakowski** [comment]
@Jay-Topher Looks all good, but seems that we missed a number of filesystem forms. I created another issue: https://warthogs.atlassian.net/browse/MAASENG-2884
 Please have a look.

Google Chrome screenshot 001667@2x
Google Chrome screenshot 001669@2x

**Review by petermakowski — APPROVED** [review]

**Comment by Jay-Topher** [comment]
> @Jay-Topher Looks all good, but seems that we missed a number of filesystem forms. I created another issue: https://warthogs.atlassian.net/browse/MAASENG-2884 Please have a look.
> 
> Google Chrome screenshot 001667@2x Google Chrome screenshot 001669@2x

I already created an issue fir this, it's next on my list

**Pull request merged**